addons/Text_Coloring/test/PrintHTMLTest.js

```diff
@@ -0,0 +1,109 @@
+TestCase("[Text_Coloring] get printable HTML", {
+    setUp: function () {
+        this.presenter = AddonText_Coloring_create();
+
+        this.model = {
+            Bottom: '',
+            ID: 'Text_Coloring1',
+            colors: [
+                {color: '#f34444', description: 'red', id: 'red'},
+                {color: '#3ba8ed', description: 'blue', id: 'blue'}
+            ],
+            text: 'Shade difference \\color{red}{red2} \\color{blue}{blue2}'
+        };
+
+        this.spies = {
+            upgradeModelSpy : sinon.spy(this.presenter, 'upgradeModel'),
+            getUserAnswerSpy : sinon.spy(this.presenter, 'getUserAnswer'),
+            createHTMLSpy : sinon.spy(this.presenter, 'createHTML')
+        };
+
+        this.userAnswer = [
+            {isSelected: true, selectionColorID: 'red', value: 'Shade'},
+            {isSelected: false, selectionColorID: null, value: 'difference'},
+            {isSelected: false, selectionColorID: null, value: 'red2'},
+            {isSelected: true, selectionColorID: 'blue', value: 'blue2'}
+        ]
+    },
+
+    'test should upgrade model when getPrintableHTML was called': function () {
+        this.presenter['printableState'] = {tokens: this.userAnswer};
+        this.presenter.getPrintableHTML(this.model, false);
+
+        assertTrue(this.spies.upgradeModelSpy.called);
+        assertTrue(this.spies.getUserAnswerSpy.called);
+        assertTrue(this.spies.createHTMLSpy.called);
+    },
+
+    'test should create printable HTML when createHTML was called': function () {
+        var htmlContent = 'Example HTML content for Text Coloring addon.';
+        var printableHTML = this.presenter.createHTML(false, htmlContent, 100, this.model);
+
+        assertTrue(printableHTML.includes(htmlContent));
+    },
+
+    'test given user answers and showAnswer on false when getPrintableHTML was called should wrap underscore word': function () {
```

Comment:
Zmieniłbym 'test should..." i "test given ... when ... should" na konwencję given when then, ale to również pierdoła.